### ISSUE TITLE
Remove register for reflection in file/ftp when upgrading to Camel 3.3

### ISSUE THREAD

**Issue by davsclaus** [apache/camel-quarkus#1113]
For example this strategy class is no longer using reflection
https://github.com/apache/camel-quarkus/blob/master/extensions/file/deployment/src/main/java/org/apache/camel/quarkus/component/file/deployment/FileProcessor.java#L38

https://issues.apache.org/jira/browse/CAMEL-14936

**Comment by davsclaus** [comment]
Not sure why GenericFile was registered for reflection too? Its an interface, maybe try to remove it or find out why

**Comment by jamesnetherton** [comment]
If you remove the reflection registration for `GenericFile` the tests pass, but there's a warning:

```
2020-04-21 13:07:29,152 WARN  [org.apa.cam.com.fil.GenericFileOnCompletion] (executor-thread-1) Error during commit. Exchange[50685704D964593-0000000000000001]. Caused by: [org.apache.camel.RuntimeCamelException - java.lang.InstantiationException: Type `org.apache.camel.component.file.GenericFile` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.]: org.apache.camel.RuntimeCamelException: java.lang.InstantiationException: Type `org.apache.camel.component.file.GenericFile` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.
	at org.apache.camel.component.file.GenericFile.copyFrom(GenericFile.java:81)
	at org.apache.camel.component.file.strategy.GenericFileExpressionRenamer.renameFile(GenericFileExpressionRenamer.java:41)
	at org.apache.camel.component.file.strategy.GenericFileRenameProcessStrategy.commit(GenericFileRenameProcessStrategy.java:114)
	at org.apache.camel.component.file.GenericFileOnCompletion.processStrategyCommit(GenericFileOnCompletion.java:132)
	at org.apache.camel.component.file.GenericFileOnCompletion.onCompletion(GenericFileOnCompletion.java:85)
	at org.apache.camel.component.file.GenericFileOnCompletion.onComplete(GenericFileOnCompletion.java:59)
	at org.apache.camel.support.UnitOfWorkHelper.doneSynchronizations(UnitOfWorkHelper.java:91)
	at org.apache.camel.impl.engine.DefaultConsumerTemplate.doneUoW(DefaultConsumerTemplate.java:223)
	at org.apache.camel.impl.engine.DefaultConsumerTemplate.receiveBodyNoWait(DefaultConsumerTemplate.java:203)
	at org.apache.camel.quarkus.component.file.it.FileResource.getFile(FileResource.java:50)

```

Looks like this is the cause:

https://github.com/apache/camel/blob/aca4ac3d1ac77321c09e717638563a417e607f55/components/camel-file/src/main/java/org/apache/camel/component/file/GenericFile.java#L79

**Comment by davsclaus** [comment]
Thanks @jamesnetherton yeah lets see if we can fix this in camel-file to avoid using that reflection based code

**Comment by davsclaus** [comment]
Okay working on a fix in upstream camel-file/camel-ftp to avoid this last reflection bits with GenericFile

**Comment by davsclaus** [comment]
Okay fixed the last reflection bits in GenericFile for Camel 3.3

**Comment by jamesnetherton** [comment]
Fixed as part of the Camel 3.3.0 upgrade.
